### DISCUSSION TITLE
Cone-shaped emitter

### DISCUSSION THREAD

**Discussion by buddhasystem** [mitsuba-renderer/mitsuba3#1225] [Q&A] [answered]
Dear All,

I'm trying to model a physics process where Cherenkov light is emitter. As a crude first approximation, I would like to experiment with a cone-shaped kind of illumination, and my intention was to use the "spot" emitter with minimum and maximum angles. I could find a reliable way to set this either in the XML code for the scene, or in the Python dictionary afterwards. Any hint or suggestions will be appreciated. Thank you!

**Comment by njroussel** [comment]
Hi @buddhasystem 

The `spot` plugin provides a `cutoff_angle` that should allow you to set a maximum angle ([documentation](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_emitters.html#spot-light-source-spot)). The documentation usually always includes a XML/Python snippet for the initialization, it typically isn't exhaustive -- for breviety not all parameters are used. But by looking at other snippets you should be able to figure out how to add a parameter of a specific type. Finally, in the documentation, each parameter labeled with the "**P**" flag in the last column will be exposed in Python with `mi.traverse()`. Parameters without that flag can only be set at initialization.
If you need to be able to set a minimum angle you'll need to either modify the source code of the `spot` emitter or write your own custom emitter.

**Comment by buddhasystem** [comment]
@njroussel Thank you! I've looked through the documentation a few times, and there is no example of setting the cutoff_angle in XML. Intuitively, this should be done withing the "rgb" tag just like the intensity, but this produces an error. I've tried a few other things but none works. Is there an example or a template of how to set this?

If I were to do this in Python, will the load from dictionary procedure yield an emitter? Can it then be used to update the scene? Thank you!

**Reply by njroussel** [reply, marked as answer]
The `cutoff_angle` is just a float parameter:
```xml
<emitter type="spot">
    <.../>
    <float name="cutoff_angle" value="30"/>
    <.../>
</emitter>
```

In Python, `mi.load_dict({'type': 'spot', ... })` will indeed return an `Emitter` object. However, it cannot be inserted into a scene. The only edits to a scene that can be done is modifying plugin parameters (marked with a "P", as mentioned in my previous comment)

**Comment by buddhasystem** [comment]
Thank you! It works great.

**Comment by buddhasystem** [comment]
@njroussel Thanks much. Very helpful.
